pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -0,0 +1,118 @@
+package cyclonedx
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	cdx "github.com/CycloneDX/cyclonedx-go"
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/handler/processor"
+)
+
+type cyclonedxParser struct {
+	rootPackage parentPackages
+}
+
+type parentPackages struct {
+	curPackage  assembler.PackageNode
+	depPackages []parentPackages
+}
+
+func NewCycloneDXParser() *cyclonedxParser {
+	return &cyclonedxParser{
+		rootPackage: parentPackages{},
+	}
+}
+
+func (c *cyclonedxParser) CreateNodes(ctx context.Context) []assembler.GuacNode {
+	nodes := []assembler.GuacNode{}
+	addNodes(c.rootPackage, &nodes)
+	return nodes
+}
+
+func addNodes(curPkg parentPackages, nodes *[]assembler.GuacNode) {
```

Comment:
just a slice instead? the slice is already a pointer type

Reply:
hmm..but we can't append with slice ?

Reply:
yea another pattern is returning the slice, append does add it to the current list most of the time but the language doesn't guarantee it. This is fine.